Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -479,17 +493,17 @@ inline bool ShouldDraw() {
         return SUCCEEDED(_firstErrorHr) && CurrentGState().ShouldDraw();
     }
 
-    inline void PushBeginDraw() {
-        if ((_beginEndDrawDepth)++ == 0) {
+    inline void PushBeginDraw(bool forceBeginDraw = false) {
+        if (forceBeginDraw || ((_beginEndDrawDepth)++ == 0)) {
             if (SUCCEEDED(_firstErrorHr)) {
                 deviceContext->BeginDraw();
             }
         }
     }
 
-    inline HRESULT PopEndDraw() {
+    inline HRESULT PopEndDraw(bool forceEndDraw = false) {
```

Comment:
could get around the flag here by pulling out the innards into some private _PopEndDraw which actually calls the end draw, which would be called by this (if beginEndDrawDepth == 1) and directly from Flush (since it already does the _beginEndDrawDepth check).